crMeta/crMetadata.go

```diff
@@ -565,31 +572,34 @@ func findCustomCRXattrFields(xattr []byte) (src, ver, pv, mv []byte, err error)
 		if err != nil {
 			return
 		}
-		if base.Equals(key, HLV_SRC_FIELD) {
+		switch string(key) {
+		case HLV_SRC_FIELD:
 			src = value
-		} else if base.Equals(key, HLV_VER_FIELD) {
+		case HLV_VER_FIELD:
 			ver = value
-		} else if base.Equals(key, HLV_PV_FIELD) {
+		case HLV_PV_FIELD:
 			pv = value
-		} else if base.Equals(key, HLV_MV_FIELD) {
+		case HLV_MV_FIELD:
 			mv = value
+		case HLV_CVCAS_FIELD:
+			cvCas = value
 		}
 	}
 	return
 }
 
-func getMetaFromMCResponse(lookupResp *base.SubdocLookupResponse) (cas uint64, cvSrc hlv.DocumentSourceId, cvVer uint64,
+func getMetaFromMCResponse(lookupResp *base.SubdocLookupResponse) (cas, cvCas uint64, cvSrc hlv.DocumentSourceId, cvVer uint64,
 	pvMap, mvMap hlv.VersionsMap, err error) {
 	cas = lookupResp.Resp.Cas
 	xattr, err := lookupResp.ResponseForAPath(base.XATTR_HLV)
 	if err != nil {
-		return 0, "", 0, nil, nil, err
+		return 0, 0, "", 0, nil, nil, err
```

Comment:
Just do a return as you have predetermined variables and err has already been set

Reply:
Done